TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/UserService.php

```diff
@@ -168,7 +168,7 @@ public function getUser($username, $authenticationProviderName = null)
         }
         $user = $this->partyService->getAssignedPartyOfAccount($account);
         if (!$user instanceof User) {
-            throw new Exception(sprintf('Unexpected user type "%s". An account with the identifier "%s" exists, but the corresponding party is not a Neos User.', get_class($user), $username), 1422270948);
+            return null;
```

Comment:
I don't know if the public API breaks, but this is necessary so that the rest of the method will be executed.